action/settings/update.go

```diff
@@ -0,0 +1,250 @@
+	"github.com/go-vela/cli/internal"
+	"github.com/go-vela/cli/internal/output"
+	"github.com/go-vela/sdk-go/vela"
+	"github.com/go-vela/server/api/types/settings"
+)
+
+// Update modifies settings based off the provided configuration.
+func (c *Config) Update(client *vela.Client) error {
+	logrus.Debug("executing update for settings configuration")
+
+	// send API call to retrieve current settings
+	s, _, err := client.Admin.Settings.Get()
+	if err != nil {
+		return err
+	}
+
+	// create the settings object
+	sUpdate := &settings.Platform{
+		Queue: &settings.Queue{
+			Routes: vela.Strings(s.GetRoutes()),
+		},
+		Compiler: &settings.Compiler{
+			CloneImage:        c.Compiler.CloneImage,
+			TemplateDepth:     c.Compiler.TemplateDepth,
+			StarlarkExecLimit: c.Compiler.StarlarkExecLimit,
+		},
+		RepoAllowlist:     vela.Strings(s.GetRepoAllowlist()),
+		ScheduleAllowlist: vela.Strings(s.GetScheduleAllowlist()),
+	}
+
+	// drop specified routes
+	if len(c.Queue.DropRoutes) > 0 {
+		newRoutes := []string{}
+
+		for _, r := range sUpdate.GetRoutes() {
+			if !slices.Contains(c.Queue.DropRoutes, r) {
+				newRoutes = append(newRoutes, r)
+			}
+		}
+
+		sUpdate.SetRoutes(newRoutes)
+	}
+
+	// add specified routes
+	if len(c.Queue.AddRoutes) > 0 {
+		routes := sUpdate.GetRoutes()
+
+		for _, r := range c.Queue.AddRoutes {
+			if !slices.Contains(routes, r) {
+				routes = append(routes, r)
+			}
+		}
+
+		sUpdate.SetRoutes(routes)
+	}
+
+	// drop specified repositories from the allowlist
+	if len(c.RepoAllowlistDropRepos) > 0 {
+		newRepos := []string{}
+
+		for _, r := range sUpdate.GetRepoAllowlist() {
+			if !slices.Contains(c.RepoAllowlistDropRepos, r) {
+				newRepos = append(newRepos, r)
+			}
+		}
+
+		sUpdate.SetRepoAllowlist(newRepos)
+	}
+
+	// add specified repositories to the allowlist
+	if len(c.RepoAllowlistAddRepos) > 0 {
+		repos := sUpdate.GetRepoAllowlist()
+
+		for _, r := range c.RepoAllowlistAddRepos {
+			if !slices.Contains(repos, r) {
+				repos = append(repos, r)
+			}
+		}
+
+		sUpdate.SetRepoAllowlist(repos)
+	}
+
+	// drop specified repositories from the allowlist
+	if len(c.ScheduleAllowlistDropRepos) > 0 {
+		newRepos := []string{}
+
+		for _, r := range sUpdate.GetScheduleAllowlist() {
+			if !slices.Contains(c.ScheduleAllowlistDropRepos, r) {
+				newRepos = append(newRepos, r)
+			}
+		}
+
+		sUpdate.SetScheduleAllowlist(newRepos)
+	}
+
+	// add specified repositories to the allowlist
+	if len(c.ScheduleAllowlistAddRepos) > 0 {
+		repos := sUpdate.GetScheduleAllowlist()
+
+		for _, r := range c.ScheduleAllowlistAddRepos {
+			if !slices.Contains(repos, r) {
+				repos = append(repos, r)
+			}
+		}
+
+		sUpdate.SetScheduleAllowlist(repos)
+	}
+
+	// manual overrides (from file)
+	if c.RepoAllowlist != nil {
+		sUpdate.RepoAllowlist = c.RepoAllowlist
+	}
+
+	if c.ScheduleAllowlist != nil {
+		sUpdate.ScheduleAllowlist = c.ScheduleAllowlist
+	}
+
+	if c.Queue.Routes != nil {
+		sUpdate.Queue.Routes = c.Queue.Routes
+	}
```

Comment:
I'm not sure if I may be missing something here, but does this function get called when there's a file input? I thought these lines from `command/settings/update.go` would prevent that: 
```go
	// check if file is provided
	if len(s.File) > 0 {
		return s.UpdateFromFile(client)
	}

	// execute the update call for the settings configuration
	//
	// https://pkg.go.dev/github.com/go-vela/cli/action/settings?tab=doc#Config.Update
	return s.Update(client)
```

Reply:
oh yeah definitely, these can be removed for sure. good catch, tweaking it

Reply:
oh wait, no, `UpdateFromFile` just prepares a Config then calls `Update`

Reply:
i suppose `UpdateFromFile` should just return a `Config` and then it should fall through to the same `Update` call you pointed out, right?